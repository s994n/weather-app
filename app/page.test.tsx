import React, { ReactNode } from "react";
import { render } from "@testing-library/react";
import Home from "./page";

jest.mock("react-dom");

// Mocking the SingleInputForm component to avoid warnings from
// specific props of children that are not recognised as HTML attributes during the snapshot test
jest.mock("@components/molecules/SingleInputForm", () => ({
  __esModule: true,
  default: ({
    children,
    action,
    inputFieldType,
    inputFieldVariant,
    buttonVariant,
    buttonAriaLabel,
    buttonLabel,
    ...props
  }: {
    children: ReactNode;
    action: (formData: FormData) => Promise<void>;
    inputFieldType: string;
    inputFieldVariant: string;
    buttonVariant: string;
    buttonAriaLabel: string;
    buttonLabel: string;
  }) => <form {...props}>{children}</form>,
}));

it("renders and matches snapshot", () => {
  const { asFragment } = render(<Home />);
  expect(asFragment()).toMatchSnapshot();
});
