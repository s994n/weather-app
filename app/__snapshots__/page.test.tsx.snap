// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <h1
      class="hero"
      style="text-transform: uppercase;"
    >
      Weather App
    </h1>
    <section
      aria-labelledby="city-current-weather-search-header"
      class="search-section"
    >
      <h2
        class="subheader"
        id="city-current-weather-search-header"
        style="text-transform: none;"
      >
        Search for the current weather by city name
      </h2>
      <form
        id="location"
        name="location"
        placeholder="City name"
      />
    </section>
  </main>
</DocumentFragment>
`;
